diff --git a/src/header/pthread/mod.rs b/src/header/pthread/mod.rs
index c742a425..a6721cad 100644
--- a/src/header/pthread/mod.rs
+++ b/src/header/pthread/mod.rs
@@ -307,6 +307,13 @@ pub unsafe extern "C" fn pthread_testcancel() {
     unsafe { pthread::testcancel() };
 }
 
+/// <https://man7.org/linux/man-pages/man3/pthread_yield.3.html>
+///
+/// Non-standard GNU extension. Prefer `sched_yield()` instead.
+pub extern "C" fn pthread_yield() {
+    let _ = Sys::sched_yield();
+}
+
 // Must be the same struct as defined in the pthread_cleanup_push macro.
 #[repr(C)]
 pub(crate) struct CleanupLinkedListEntry {
