diff --git a/src/disk/file.rs b/src/disk/file.rs
index 78d51bc..f923d72 100644
--- a/src/disk/file.rs
+++ b/src/disk/file.rs
@@ -43,10 +43,12 @@ impl<T> ResultExt for std::io::Result<T> {
 
 impl DiskFile {
     pub fn open(path: impl AsRef<Path>) -> Result<DiskFile> {
+        let path = path.as_ref();
         let file = OpenOptions::new()
             .read(true)
             .write(true)
             .open(path)
+            .or_else(|_| OpenOptions::new().read(true).open(path))
             .or_eio()?;
         Ok(DiskFile { file })
     }
