--- a/dbus-1.16.2/bus/system.conf.in
+++ b/dbus-1.16.2/bus/system.conf.in
@@ -82,23 +82,23 @@
   </policy>
 
   <!-- Only systemd, which runs as root, may report activation failures. -->
-  <policy user="root">
+  <policy user="0">
     <allow send_destination="org.freedesktop.DBus"
            send_interface="org.freedesktop.systemd1.Activator"/>
   </policy>
 
   <!-- root may monitor the system bus. -->
-  <policy user="root">
+  <policy user="0">
     <allow send_destination="org.freedesktop.DBus"
            send_interface="org.freedesktop.DBus.Monitoring"/>
   </policy>
 
   <!-- If the Stats interface was enabled at compile-time, root may use it.
        Copy this into system.local.conf or system.d/*.conf if you want to
        enable other privileged users to view statistics and debug info -->
-  <policy user="root">
+  <policy user="0">
     <allow send_destination="org.freedesktop.DBus"
            send_interface="org.freedesktop.DBus.Debug.Stats"/>
   </policy>
