diff --git a/drivers/common/src/logger.rs b/drivers/common/src/logger.rs
index 82ec2bd0..a531edd9 100644
--- a/drivers/common/src/logger.rs
+++ b/drivers/common/src/logger.rs
@@ -44,6 +44,7 @@ pub fn setup_logging(
         Ok(b) => {
             logger = logger.with_output(b.with_filter(file_level).flush_on_newline(true).build())
         }
+        Err(error) if error.raw_os_error() == Some(19) => {}
         Err(error) => eprintln!("Failed to create {logfile_base}.log: {}", error),
     }
 
@@ -61,6 +62,7 @@ pub fn setup_logging(
                     .build(),
             )
         }
+        Err(error) if error.raw_os_error() == Some(19) => {}
         Err(error) => eprintln!("Failed to create {logfile_base}.ansi.log: {}", error),
     }
 
