--- /dev/null
+++ b/init.d/29_activate_console.service
@@ -0,0 +1,8 @@
+[unit]
+description = "Activate console VT"
+requires_weak = ["00_base.target"]
+
+[service]
+cmd = "inputd"
+args = ["-A", "2"]
+type = "oneshot_async"
--- /dev/null
+++ b/init.d/30_console.service
@@ -0,0 +1,8 @@
+[unit]
+description = "Console getty on VT2"
+requires_weak = ["29_activate_console.service"]
+
+[service]
+cmd = "getty"
+args = ["2"]
+type = "oneshot_async"
--- /dev/null
+++ b/init.d/31_debug_console.service
@@ -0,0 +1,8 @@
+[unit]
+description = "Debug console on VT3"
+requires_weak = ["29_activate_console.service"]
+
+[service]
+cmd = "getty"
+args = ["3"]
+type = "oneshot_async"
